rest_server/src/controllers/item_controller.js

```diff
@@ -246,7 +259,15 @@ const listCategories = asyncHandler(async (req, res, next) => {
 const addCategory = asyncHandler(async (req, res, next) => {
   try {
     let result = await MarketplaceItem.get(req.params.itemId);
-    if (checkReadPermission(req.userInfo, result)) {
+    if (await checkWritePermission(req.tokenInfo, result)) {
+      const category = await ItemCategory.get(req.params.categoryId);
+      if (category.name === OFFICIAL_EXAMPLE && req.tokenInfo.admin !== true) {
+        return next(
+          error.createForbidden(
+            'Only admin can set "official example" category.',
```

Comment:
>official example [](http://example.com/codeflow?start=33&length=16)

should we use the const varable here?

Reply:
Updated, change to `Only admin can set "${ItemCategory.OFFICIAL_EXAMPLE}" category.`